Packages/com.unity.inputsystem/InputSystem/Editor/PropertyDrawers/InputActionDrawerBase.cs

```diff
@@ -38,21 +50,56 @@ private void InitTreeIfNeeded(SerializedProperty property)
             //       changing behind our backs by undo/redo here. Being a PropertyDrawer, we will automatically
             //       get recreated by Unity when it touches our serialized data.
 
-            if (m_TreeView == null)
+            var viewData = GetOrCreateViewData(property);
+            var propertyIsClone = IsPropertyAClone(property);
+
+            if (viewData.TreeView != null && !propertyIsClone)
+                return;
+            
+            if (propertyIsClone)
             {
-                // Create tree and populate it.
-                m_TreeView = new InputActionTreeView(property.serializedObject)
-                {
-                    onBuildTree = () => BuildTree(property),
-                    onDoubleClick = OnItemDoubleClicked,
-                    title = (property.displayName, property.GetTooltip()),
-                    // With the tree in the inspector, the foldouts are drawn too far to the left. I don't
-                    // really know where this is coming from. This works around it by adding an arbitrary offset...
-                    foldoutOffset = 14,
-                    drawActionPropertiesButton = true
-                };
-                m_TreeView.Reload();
+                property.FindPropertyRelative(nameof(InputAction.m_Id)).stringValue = "";
+                property.FindPropertyRelative(nameof(InputAction.m_Name)).stringValue = "Input Action";
+                property.FindPropertyRelative(nameof(InputAction.m_SingletonActionBindings)).ClearArray();
+                property.serializedObject.ApplyModifiedProperties();
```

Comment:
Probably better to use `ApplyModifiedPropertiesWithoutUndo` here.

Reply:
:+1: